src/v/utils/retry_chain_node.cc

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/retry_chain_node.h"
+
+#include "ssx/sformat.h"
+#include "vassert.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/ragel.hh>
+
+#include <limits>
+#include <variant>
+
+using namespace std::chrono_literals;
+
+/// Jitter
+thread_local static fast_prng _retry_rand_jitter_source;
```

Comment:
we have somethign similar in the `random/utils.h` i do think that library is missing the `fast_prng` thread-local so you can make this an alias there too. 

`fast_prng_source()` method or smth there and import it.